bundles/org.openhab.ui.basic/web-src/smarthome.js

```diff
@@ -1601,17 +1655,51 @@
 
 			var newValue = value;
 			var undefValue = "";
-			if (itemState === "undefined" || itemState === "NULL" || itemState === "UNDEF") {
+			if (itemState === undefined) {
+				undefValue = lastUndef;
+			} else if (itemState === "NULL" || itemState === "UNDEF") {
 				newValue = "";
-				undefValue = value;
+				undefValue = (value !== "") ? value : lastUndef;
 			}
 
+			if (_t.inputHint === "number") {
+				if (newValue !== "") {
+					newValue = parseNumber(newValue).value;
+					var valueArray = newValue.trim().split(" ");
+					if (valueArray.length > 0) {
+						newValue = valueArray[0];
+					}
+					if (valueArray.length > 1) {
+						_t.input.parentNode.nextElementSibling.innerHTML = valueArray[1];
+					}
+				} else {
+					var undefArray = undefValue.split(" ");
+					if (undefArray.length > 0) {
+						undefValue = undefArray[0];
+					}
+				}
+			} else if (_t.itemType === "datetime") {
+				newValue = ((lastItemState !== "NULL") && (lastItemState !== "UNDEF")) ? itemState : value;
+				newValue = newValue.trim().split(".")[0];		// drop millis
```

Comment:
The problem is here.
It is possible that value does not contain a "." (state pattern defined as `[%s]`):
![image](https://user-images.githubusercontent.com/10186704/236819403-aac17295-35f4-4f34-9433-a224bb39326d.png)
Is there a reason to not consider only itemState here ?
Are you trying to consider the state pattern defined for the item ? Is it a good idea for date& time ?

Reply:
I was not understanding that line ;)

Reply:
Already fixed, I came to the same conclusion, only using itemState for datetime.

Reply:
Apparantly a %s state pattern drops the millis (but still shows the time offset). The events always have millis.
Note that in all the logic, I drop the time offset. I send without offset and drop the offset from received values.